raiden/network/proxies/token_network.py

```diff
@@ -1032,36 +997,67 @@ def close(
 
         log.info('closeChannel successful', **log_details)
 
-    def _update_preconditions(
+    def update_transfer(
             self,
             channel_identifier: ChannelID,
             partner: Address,
             balance_hash: BalanceHash,
             nonce: Nonce,
             additional_hash: AdditionalHash,
             closing_signature: Signature,
-            block_identifier: BlockSpecification,
-    ) -> None:
-        if not self.client.can_query_state_for_block(block_identifier):
```

Comment:
This check is completely bugous. It may be the case that at the time of the call to `can_query_state_for_block` the block is not pruned, but on the call `_check_for_outdated_channel` bellow a new block is mined, forcing the current block to be pruned. The **only** way to handle this is to rely on the json rpc response raising an exception and handling it